### PR TITLE
builtin property: accept keyword arguments

### PR DESCRIPTION
**Pull request by chrysn**
alternative pull request to #1779 if you prefer less fine-grained commits

### CODE REVIEW

File: py/objproperty.c [hunk 1]
```diff
@@ -38,28 +38,24 @@ typedef struct _mp_obj_property_t {
...
-        o->proxy[1] = mp_const_none;
-    }
-    if (n_args >= 1) {
-        o->proxy[0] = args[0];
-    } else {
-        o->proxy[0] = mp_const_none;
-    }
+    o->proxy[ARG_fget] = vals[0].u_obj;
+    o->proxy[ARG_fset] = vals[1].u_obj;
+    o->proxy[ARG_fdel] = vals[2].u_obj;
```
Comment (dpgeorge) [inline py/objproperty.c]: It should be `o->proxy[0] = vals[ARG_fget].u_obj` etc.

File: py/objproperty.c [hunk 2]
```diff
@@ -38,28 +38,24 @@ typedef struct _mp_obj_property_t {
...
+        { MP_QSTR_, MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+        { MP_QSTR_doc, MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+    };
+    mp_arg_val_t vals[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, args, MP_ARRAY_SIZE(allowed_args), allowed_args, vals);
 
     mp_obj_property_t *o = m_new_obj(mp_obj_property_t);
     o->base.type = type;
-    if (n_args >= 4) {
+    if (vals[ARG_doc].u_obj != mp_const_none) {
```
Comment (dpgeorge) [inline py/objproperty.c]: Remove this line (and empty {} block) and simply add a comment after the assignment of ARG_fdel that the doc arg is ignored.  It's simpler and still obvious what's going on.

**Comment by dpgeorge** [comment]
One single commit is appropriate for this sized patch.  Thanks!

**Comment by pfalcon** [comment]
I merged this, only then noticing that commit message doesn't follow standard uPy practices. @chrysn, please follow them next time (just doing git log would give good idea what they're, but just in case: the prefix of subj line would be "py/objproperty:", there should be grammatical sentences (first letter capitalized, not just full stop at the end.)) Thanks.

**Comment by pfalcon** [comment]
And feature-per-commit-don't-put-unrelated-things-in-one-commit commits are appreciated, but here it's indeed one commit is just right.

**Pull request closed without merging**